src/tests/server.mjs

```diff
@@ -151,6 +153,127 @@ router.put("/messages/:id", (request, response) => {
   }
 })
 
+router.get("/posts", async (request, response) => {
+  const { worker_id } = request.query
+
+  const posts = await getPosts(worker_id)
+
+  const res = templateRenderer.render("./posts", { posts, worker_id })
+
+  response.type("html").status(200).send(res)
+})
+
+router.post("/posts", async (request, response) => {
+  const { title, body, worker_id } = request.body
+
+  await addPost(title, body, worker_id)
+
+  response.redirect(303, `/__turbo/posts?worker_id=${worker_id}`)
+})
+
+router.get("/posts/:id/", async (request, response) => {
+  const { worker_id } = request.query
+  const { id } = request.params
+
+  const posts = await getPosts(worker_id)
+  const post = posts.find((post) => post.id == id)
+
+  if (post) {
+    const res = templateRenderer.render("./post", { post, worker_id })
+
+    response.type("html").status(200).send(res)
+  } else {
+    response.sendStatus(404)
+  }
+})
+
+router.post("/posts/:id", async (request, response) => {
+  if (request.body._method == "delete") {
+    const { worker_id } = request.body
+    const { id } = request.params
+
+    await deletePost(id, worker_id)
+    response.redirect(303, `/__turbo/posts?worker_id=${worker_id}`)
+  } else {
+    response.sendStatus(422)
+  }
+})
+
+router.post("/posts/:post_id/comments", async (request, response) => {
+  const { post_id } = request.params
+  const { worker_id } = request.body
+  const { body } = request.body.comment
+
+  const post = await getPost(post_id, worker_id)
+
+  if (post) {
+    await addComment(post_id, body, worker_id)
+
+    response.redirect(303, `/__turbo/posts/${post_id}?worker_id=${worker_id}`)
+  } else {
+    response.sendStatus(404)
+  }
+})
+
+const postsDatabaseName = (worker_id) => {
+  return `src/tests/fixtures/volatile_posts_database_${worker_id}.json`
+}
+
+const ensureDatabase = async (worker_id) => {
+  if (worker_id == null || worker_id == undefined) {
+    throw new Error("worker_id is required")
+  }
+
+  if (!fs.existsSync(postsDatabaseName(worker_id))) {
+    fs.writeFileSync(postsDatabaseName(worker_id), "[]")
+  }
+}
+
```

Comment:
I'm using `async` even when using synchronous functions, just in case we ever decide to use some other database implementation, like sqlite. In such case, we'd only modify these reader/writer functions.